starboard/common/spin_lock.cc

```diff
@@ -18,16 +18,17 @@
 
 namespace starboard {
 
-void SpinLockAcquire(SbAtomic32* atomic) {
-  while (SbAtomicAcquire_CompareAndSwap(atomic, kSpinLockStateReleased,
-                                        kSpinLockStateAcquired) ==
-         kSpinLockStateAcquired) {
+void SpinLockAcquire(std::atomic_int* atomic) {
```

Comment:
atomic_int32_t?

Reply:
atomic_int32_t is optional type and might be not available for some platforms. I used atomic<int32_t> instead.